### PR TITLE
AI Assistant home automation system

### PR DESCRIPTION
**Pull request by innovativecreations**
## Describe your project in 1-3 sentences
My aim is to build an AI assistant (like Google home, Alexa) along with hardware to control my appliances such as light, ac, etc, with raspberry pi, esp32, relays, etc. The next step is to build an AI assistant.

## Checklist before submitting proposal:

- [x] I am a current high school student (or younger).
- [x] I have submitted an RSVP at <https://hackclub.com/winter>
- [x] I am a member of the Hack Club Slack (Join at <https://hackclub.com/slack>)
- [x] I have self-reviewed my project proposal.

### CODE REVIEW

**Comment by bellesea** [comment]
heyy thanks so much for making a PR! we're back on tuesday and will start reviewing PRs in the order that they came on. 

to make the process smoother, please ensure you double checked that your file is properly formatted, you have all the parts you need, and that your plan is detailed enough for us to imagine how you're using the components in your budget to build your project and visualise you work on it :)

and fyi, we're looking for projects where/that:
- you can spend at least 10 days building (from Feb 14th to 23rd)
- you are able to modify or make your own in some ways
- you can learn something new from
- involves working with electronics or software on a lower level than a pre-assembled kit of parts

and, we normally don’t approve projects where you buy consumer components w/o modification like keyboards (buying PCB, keycaps, switches, and assembling it), PCs (buying motherboard, cpu, memory and plugging it together), and kits (eg. closed-sourced consumer drone kits). 

_(note: from a first glance of your project i'm wondering: what's the reason for the huge monitor? does your budget include shipping and taxes?)_

our Jan 15 11:59pm ET deadline is for PRs to come in and not for final approval so we'll work with you to either accept this project OR give you a chance to submit another PR. 

if you have any questions, please do not hesitate to ask her or DM me (i'm belle on slack), thank you!

**Comment by bellesea** [comment]
@innovativecreations thanks for the additional context! a heads up that we've started reviewing projects but due to the flurry last weekend (100+ PRs dsfsdfgfdg) it might take a while before we review yours, thank you for your patience!

File: innovativecreations.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
+name: "@innovativecreations"
+project: "AI assistant cum home automation system"
+---
+
+# AI Assistant cum home automation system
```
Comment (cfanoulis) [inline innovativecreations.md:6]:
```suggestion
# AI Assistant home automation system
```

**Comment by cfanoulis** [comment]
Also, please make sure you fill out the PR template - we need your slack username to get in touch :)

File: innovativecreations.md (hunk 1 above)
Comment (innovativecreations) [reply innovativecreations.md:6]: Done

File: innovativecreations.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+---
+name: "@innovativecreations"
+project: "AI assistant cum home automation system"
```
[the remark this answers is not in the document]
Comment (innovativecreations) [reply innovativecreations.md]: Done

**Comment by innovativecreations** [comment]
> Also, please make sure you fill out the PR template - we need your slack username to get in touch :)

Slack username - @mayank

**Review by cfanoulis — DISMISSED** [review]
thanks for the update. some more things
 - i see you want to have this voice-controlled, but I don't see any microphones included, maybe you need to go shopping for one?
 - i see you'll be 3d printing your case - do you have enough filament to do so?
 - what kind of appliances would you be controlling? are esps & relays enough for them?
 - how's the LED strip being used here?
 - Why do you need both a hard disk & a SD Card?
 - Do you have any more reference material? This YouTube video was not very clear, and I didn't see any step-by-step guide linked.
 - Have you factored shipping & taxes in your total budget?

**Comment by innovativecreations** [comment]
> 
• Mic is included in the pi accessories
• Yes, I've sufficient filament
• I want to control the light bulbs, fan, power sockets, led strip, etc; Actually I forgot to mention the Arduino(for controlling the led strip)
• The led of the strips(WS2812b) can be individually controlled, so I want to use it as an indication of switching the work modes for example:- I'll use blue for coding, warm white for studying, etc.
• I'm using the sd card for the OS files and the hard drive for the codes and data so that in case the os crashes the data will be safe.
• Actually the video was just for understanding the basic principle of the project, I'm integrating many things into one.
   "https://www.youtube.com/watch?v=EJaWXQiDQcw" for controlling appliances but instead of esp8266 I'm using esp32 and instead of MIT app inventor, I'll build the app in flutter or Kotlin.
 "https://www.youtube.com/watch?v=aO92B-K4TnQ" for interfacing the esp32 to firebase.
• The shipping is free.

**Comment by cfanoulis** [comment]
Thanks for that - can you please update your budget to include every item in a single line (ie raspi on one line, microphone on another)? Items you already own can have a price of 0$

**Comment by innovativecreations** [comment]
> Thanks for that - can you please update your budget to include every item in a single line (ie raspi on one line, microphone on another)? Items you already own can have a price of 0$

done.

**Comment by cfanoulis** [comment]
Thanks for including those. Will you be writing your own software for this, or are you using open-source for this?

**Comment by innovativecreations** [comment]
> Thanks for including those. Will you be writing your own software for this, or are you using open-source for this?

I'm writing my own code

**Comment by cfanoulis** [comment]
Hi, sz for the delay! This is really delayed and it's my fault - I'll be reviewing it in the next 24h. I fI don't, pls yell at me in Slack, thx <3

**Comment by jianmin-chen** [comment]
Hey @innovativecreations! I'm looking over your PR. It looks great, but Char is right - a bit more detail would be nice. Can you give a bit more context as to how Firebase will be factored in?

**Comment by innovativecreations** [comment]
> Hey @innovativecreations! I'm looking over your PR. It looks great, but Char is right - a bit more detail would be nice. Can you give a bit more context as to how Firebase will be factored in?

I'm using the Real-time database to monitor and change the status of the appliances.

**Comment by innovativecreations** [comment]
> > Hey @innovativecreations! I'm looking over your PR. It looks great, but Char is right - a bit more detail would be nice. Can you give a bit more context as to how Firebase will be factored in?
> 
> I'm using the Real-time database to monitor and change the status of the appliances.

**Comment by jianmin-chen** [comment]
Okay! Are these devices already IOT? I'm asking because it seems as if you haven't clarified how exactly you're planning to connect these devices to the Internet.

**Comment by innovativecreations** [comment]
> Okay! Are these devices already IOT? I'm asking because it seems as if you haven't clarified how exactly you're planning to connect these devices to the Internet.

Yes, the Esp32 comes with Wi-Fi and BLE

**Comment by jianmin-chen** [comment]
Okay! I know that the ESP32 has WiFi and Bluetooth; I want to make sure that you've made sure that the devices you're planning to connect (AC, lights, etc.) have those functionalities; if you could just make a short list of devices you're planning to connect that would be useful!

**Comment by innovativecreations** [comment]
> Okay! I know that the ESP32 has WiFi and Bluetooth; I want to make sure that you've made sure that the devices you're planning to connect (AC, lights, etc.) have those functionalities; if you could just make a short list of devices you're planning to connect that would be useful!

I'll use relay to operate the normal AC appliance like light-bulbs, power socket

**Comment by innovativecreations** [comment]
> Okay! I know that the ESP32 has WiFi and Bluetooth; I want to make sure that you've made sure that the devices you're planning to connect (AC, lights, etc.) have those functionalities; if you could just make a short list of devices you're planning to connect that would be useful!

For reference :- https://youtube.com/watch?v=EJaWXQiDQcw&feature=shares

**Review by jianmin-chen — APPROVED** [review]
Alright, awesome! @hackclub/bank-ops will be reaching out soon with more info :)

**Review by daisybanks — APPROVED** [review]
Hi @innovativecreations! Everything here is looking and sounding good! I'm excited to see you ship this! I'm dming you now on Slack to talk about the next steps in getting you your grant!

**Comment by innovativecreations** [comment]
> Hi @innovativecreations! Everything here is looking and sounding good! I'm excited to see you ship this! I'm dming you now on Slack to talk about the next steps in getting you your grant!

**Comment by bellesea** [comment]
hey @innovativecreations did you mean to close the PR?

**Comment by innovativecreations** [comment]
> hey @innovativecreations did you mean to close the PR?

I was just hover over the github, I don't mean to close the PR

**Comment by bellesea** [comment]
no worries, just reopened it :)

**Comment by innovativecreations** [comment]
> no worries, just reopened it :)

Thanks

**Comment by innovativecreations** [comment]
> Hi @innovativecreations! Everything here is looking and sounding good! I'm excited to see you ship this! I'm dming you now on Slack to talk about the next steps in getting you your grant!

@Mayank on slack

**Pull request merged**